### ISSUE TITLE
net init problem

### ISSUE THREAD

**Issue by jingpengwu** [seung-lab/znn-release#97]
compiled locally and get the following error:
```
cmd: python train.py -c config.cfg 
error msg : 
initialize a new network...
Traceback (most recent call last):
 File "train.py", line 263, in <module>
   main( args )
 File "train.py", line 69, in main
   net, lc = znetio.create_net(pars)
 File "/cnnpkg/znn-release/python/front_end/znetio.py", line 457, in create_net
   net = init_network( pars )
 File "/cnnpkg/znn-release/python/front_end/znetio.py", line 445, in init_network
   _num_threads, _optimize, phase, _force_fft)
Boost.Python.ArgumentError: Python argument types in
   CNet.__init__(CNet, str, numpy.ndarray, int, bool, int, bool)
did not match C++ signature:
   __init__(boost::python::api::object, boost::python::tuple, std::string, boost::numpy::ndarray, unsigned long, bool, unsigned char, bool)
   __init__(boost::python::api::object, std::string, boost::numpy::ndarray, unsigned long, bool, unsigned char, bool)
```

the first errors were gone after changing string types into char* in C++ code.

```
cmd: python train.py -c config.cfg 
error msg : 
initialize a new network...
Traceback (most recent call last):
 File "train.py", line 263, in <module>
   main( args )
 File "train.py", line 69, in main
   net, lc = znetio.create_net(pars)
 File "/cnnpkg/znn-release/python/front_end/znetio.py", line 457, in create_net
   net = init_network( pars )
 File "/cnnpkg/znn-release/python/front_end/znetio.py", line 445, in init_network
   _num_threads, _optimize, phase, _force_fft)
Boost.Python.ArgumentError: Python argument types in
   CNet.__init__(CNet, str, numpy.ndarray, int, bool, int, bool)
did not match C++ signature:
   __init__(boost::python::api::object, boost::python::tuple, std::string, boost::numpy::ndarray, unsigned long, bool, unsigned char, bool)
   __init__(boost::python::api::object, std::string, boost::numpy::ndarray, unsigned long, bool, unsigned char, bool)
```

**Comment by jinseopkim** [comment]
after the change, now the error is this. still the error seems to be related to interfacing the string type variable between python and C++

```
create input image class...
boundary mirror...

create label image class...
start training...
start from  1
save as  ../jwtest/net_init.h5
stdpre:  /processing/znn/train/statistics/
Traceback (most recent call last):
 File "train.py", line 263, in <module>
   main( args )
 File "train.py", line 130, in main
   props = net.forward( vol_ins )
TypeError: No registered converter was able to produce a C++ rvalue of type std::string from this Python object of type str
```

**Comment by torms3** [comment]
The problem was easily solved by upgrading GCC from v4.9.3 to v5.3.1.
It is weird though why the older GCC 4.8 doesn't have error, 4.9 does, and 5.3 doesn't.
